packages/react/src/BaseChat/BaseChat.test.tsx

```diff
@@ -0,0 +1,338 @@
+import { ProfileResource, createReference, getReferenceString } from '@medplum/core';
+import { Bundle, Communication } from '@medplum/fhirtypes';
+import { BartSimpson, DrAliceSmith, HomerSimpson, MockClient } from '@medplum/mock';
+// @ts-expect-error _subscriptionController is not exported from module normally
+import { MedplumProvider, _subscriptionController } from '@medplum/react-hooks';
+import crypto from 'node:crypto';
+import { EventEmitter } from 'node:events';
+import { useState } from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { act, fireEvent, render, screen } from '../test-utils/render';
+import { BaseChat, BaseChatProps } from './BaseChat';
+
+type TestComponentProps = Omit<Omit<BaseChatProps, 'communications'>, 'setCommunications'>;
+
+const homerReference = createReference(HomerSimpson);
+const homerReferenceStr = getReferenceString(homerReference);
+const drAliceReference = createReference(DrAliceSmith);
+const drAliceReferenceStr = getReferenceString(drAliceReference);
+const HOMER_DR_ALICE_CHAT_QUERY = `sender=${homerReferenceStr},${drAliceReferenceStr}&recipient=${homerReferenceStr},${drAliceReferenceStr}`;
+
+jest.mock('@medplum/react-hooks', () => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const EventEmitter = require('node:events');
```

Comment:
Does this have to be `EventEmitter` from `node:events`?

Could we use the `EventTarget` from `@medplum/core`?